downloadDirectory/src/main/java/org/reactome/release/downloadDirectory/Biopax.java

```diff
@@ -0,0 +1,211 @@
+				outfile = "biopax.zip";
+			}
+			validateBioPAX(biopaxDir, biopaxLevel, releaseNumber, outfile);
+			// Move zip file and remove temp directory
+			String outpathBiopax = releaseNumber + "/" + outfile;
+			Files.move(Paths.get(outfile), Paths.get(outpathBiopax), StandardCopyOption.REPLACE_EXISTING);
+			FileUtils.deleteDirectory(new File(biopaxDir));
+		}	
+  	}
+  	
+    //Generate BioPAX files using the appropriate SpeciesAllPathwaysConverter function found in Pathway-Exchange
+	private static void generateBioPAXFile(String host, String database, String username, String password, String port, String biopaxDir, String speciesName, int biopaxLevel) throws Exception {
+		
+		if (biopaxLevel == 2) {
+			SpeciesAllPathwaysConverter converter = new SpeciesAllPathwaysConverter();
+			converter.doDump(new String[]{host, database, username, password, port, biopaxDir, speciesName});			
+		} else if (biopaxLevel == 3) {
+			SpeciesAllPathwaysLevel3Converter level3converter = new SpeciesAllPathwaysLevel3Converter();
+			level3converter.doDump(new String[]{host, database, username, password, port, biopaxDir, speciesName});
+		}
+	}
+	
+	// Next, once all species files for the current BioPAX level have been generated, we must validate them to make sure they are appropriately formatted
+	private static void validateBioPAX(String biopaxDir, int biopaxLevel, String releaseNumber, String outfile) throws Exception {
+		
+		// Rename files, replacing whitespace with underscores
+	    Files.newDirectoryStream(Paths.get(biopaxDir),
+	    	      owlFilepath -> owlFilepath.toString().endsWith(".owl"))
+	    	      .forEach(owlFile -> {
+	    	          String owlFilename = owlFile.toFile().getPath();
+	    	          owlFilename = owlFilename.replaceAll(" +", "_");
+	    	          File formattedOwlFilename = new File(owlFilename);
+	    	          owlFile.toFile().renameTo(formattedOwlFilename);
+	    	      });
+	    // Validate each owl file in the biopax output directory produced by PathwaysConverter
+	    logger.info("Validating owl files...");
+		runBiopaxValidation(biopaxDir);
+
+		// Compress all Biopax and validation files into individual zip files
+		logger.info("Zipping BioPAX" + biopaxLevel + " files...");
+		ZipOutputStream biopaxZipStream = getBiopaxZipStream(biopaxLevel);
+		ZipOutputStream validatorZipStream = getValidatorZipStream(biopaxLevel);
+		Files.newDirectoryStream(Paths.get(biopaxDir),
+			      path -> (path.toString().endsWith(".xml") || path.toString().endsWith(".owl")))
+			      .forEach(f -> {
+			          String spath = f.toFile().getPath();
+			          File file = new File(spath);
+		        	  try {
+		    	          if (spath.endsWith(".owl")) {
+		    	        	writeToZipFile(file, biopaxZipStream);
+		    	          } else if (spath.endsWith(".xml")) {
+							writeToZipFile(file, validatorZipStream);
+		    	          }
+					  } catch (IOException e) {
+						e.printStackTrace();
+					  }
+			      });
```

Comment:
Maybe the directory stream writing to the zip file can be shortened like this:

```
Files.newDirectoryStream(Paths.get(biopaxDir), path -> (path.toString().endsWith(".xml")))
    .forEach(f -> writeToZipFile(f.toFile(), validatorZipStream));
Files.newDirectoryStream(Paths.get(biopaxDir), path -> (path.toString().endsWith(".owl")))
    .forEach(f -> writeToZipFile(f.toFile(), biopaxZipStream));
```

The writeToZipFile method body would just have to be surrounded by a try instead of throwing the IOException.

Reply:
You could actually extract the common code and do this:

```
writeFilesToZipFile(biopaxDir, ".xml", validatorZipStream);
writeFilesToZipFile(biopaxDir,".owl", biopaxZipStream);

private static void writeFilesToZipFile(String biopaxDir, String fileExtension, ZipOutputStream zipOutputStream) throws IOException {
    Files.newDirectoryStream(Paths.get(biopaxDir), path -> path.toString().endsWith(fileExtension))
         .forEach(path -> writeToZipFile(path.toFile(), zipOutputStream));
}
```

Reply:
This is nice and succinct, when I get to resolving this PR I'll look into it